### PR TITLE
KAFKA-4659: Improve test coverage of CachingKeyValueStore

### PR DESCRIPTION
**Pull request by jeyhunkarimov**

### CODE REVIEW

**Comment by jeyhunkarimov** [comment]
call for review, @mjsax @dguy

File: streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java [hunk 1]
```diff
@@ -243,6 +245,50 @@ public void shouldThrowIfTryingToDoPutIfAbsentClosedCachingStore() throws Except
         store.putIfAbsent("b", "c");
     }
 
+    @Test
+    public void shouldThrowNullPointerException() {
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java]: this should be three test IMHO

**Review by mjsax — COMMENTED** [review]
Thanks for the patch!

**Review by mjsax — APPROVED** [review]
LGTM.

Call for final review and merging @guozhangwang

**Comment by guozhangwang** [comment]
LGTM. Would like to leave to @dguy for a final review.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java [hunk 2]
```diff
@@ -243,6 +245,50 @@ public void shouldThrowIfTryingToDoPutIfAbsentClosedCachingStore() throws Except
...
+        assertTrue(store.get("b").equals("2"));
+
+        store.putIfAbsent("b", "3");
+        assertTrue(store.get("b").equals("2"));
+    }
+
+    @Test
+    public void shouldPutAll() {
+        List<KeyValue<String, String>> entries = new ArrayList<>();
+        entries.add(new KeyValue<String, String>("a", "1"));
```
Comment (dguy) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java]: nit: can remove type arguments

File: streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java [hunk 3]
```diff
@@ -243,6 +245,50 @@ public void shouldThrowIfTryingToDoPutIfAbsentClosedCachingStore() throws Except
         store.putIfAbsent("b", "c");
     }
 
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerExceptionOnPutWithNullKey() {
+        store.put(null, "c");
```
Comment (dguy) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java:250]: The `CachingKeyValueStore` doesn't throw an NPE here, rather `org.apache.kafka.common.utils.Bytes.LexicographicByteArrayComparator.compare` does. We probably should add `Objects.requireNonNull(...)` to `CachingKeyValueStore#put(..)` etc

**Review by dguy — COMMENTED** [review]
Thanks @jeyhunkarimov just a couple of comments

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5559/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5572/
Test PASSed (JDK 7 and Scala 2.11).

**Review by dguy — APPROVED** [review]
Thanks @jeyhunkarimov LGTM

**Pull request closed without merging**